### PR TITLE
Add supports for hapi 17

### PR DESCRIPTION
**Pull request by joseluisdiaz**
Added support to the factory method to return the right object given the HAPI version. To run the test, `dev` dependencies are a specific feature in `yarn`. Tu run the test, please use `yarn` :)

### CODE REVIEW

File: eventlogger/hapi_server_v17.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
+var assign = require('lodash').assign;
+
+module.exports.watch = function (logger, server, options) {
+  options = assign({}, {
+    ignorePaths: []
+  }, options);
+
+  var ignorePaths = options.ignorePaths;
```
Comment (santiagoaguiar) [inline eventlogger/hapi_server_v17.js]: Minor: use `const`

File: test/hapi_server.js [hunk 2]
```diff
@@ -56,11 +56,20 @@ describe('watch Hapi server', function () {
     });
   });
 
+  after(function (done) {
+    server.stop(done);
+  });
+
   it('should log response time', function (done) {
     eventLogger.logger.info = function(log_event) {
+      if (log_event.log_type === 'request') {
```
Comment (santiagoaguiar) [inline test/hapi_server.js]: why the changes on these tests?

Comment (joseluisdiaz) [reply test/hapi_server.js]: Not sure why it works before! in the `request` is when the clock starts.

File: eventlogger/hapi_server_v17.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    return h.continue;
+  }
+
+  function onRequestError(request, error) {
+    logger.error({
+      log_type: 'request_error',
+      req: request,
+      res: request.response,
+      err: error,
+      payload: request.pre &&
```
Comment (dafortune) [inline eventlogger/hapi_server_v17.js]: Minor, The indentation looks confusing to me.

File: eventlogger/hapi_server_v17.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+      request.pre._originalPayload ||
+      request.payload
```
Comment (dafortune) [inline eventlogger/hapi_server_v17.js]: I know it was already like that but, is it safe to log the payload like this? what if there were passwords or things like those?

File: eventlogger/index.js [hunk 5]
```diff
@@ -17,7 +18,13 @@ EventLogger.prototype.watch = function (obj, options) {
 
   //stupid way of identify a hapi server
   if (obj.app && obj.info && obj.info.id && obj.info.created) {
-    return hapi_server.watch(this.logger, obj, options);
+    const hapiVersion = obj.version || ""; // '17.0.0'
+    const majorHapiVersion = parseInt(hapiVersion.split(/\./)[0]);
+
+    // depends on the hapi version use a different watch version
```
Comment (dafortune) [inline eventlogger/index.js]: s/depends/depending

File: eventlogger/index.js [hunk 6]
```diff
@@ -17,7 +18,13 @@ EventLogger.prototype.watch = function (obj, options) {
 
   //stupid way of identify a hapi server
   if (obj.app && obj.info && obj.info.id && obj.info.created) {
-    return hapi_server.watch(this.logger, obj, options);
+    const hapiVersion = obj.version || ""; // '17.0.0'
```
Comment (dafortune) [inline eventlogger/index.js]: I'm still ambivalent about this multiple version support, but seems to be the way the lib is doing it for other tools, so ok.

File: eventlogger/index.js [hunk 7]
```diff
@@ -17,7 +18,13 @@ EventLogger.prototype.watch = function (obj, options) {
...
+    const majorHapiVersion = parseInt(hapiVersion.split(/\./)[0]);
```
Comment (dafortune) [inline eventlogger/index.js]: Extremely minor: replacing the regexp by a '.' string might be clearer.

File: package.json [hunk 8]
```diff
@@ -9,10 +9,11 @@
   "author": "",
   "license": "ISC",
   "devDependencies": {
-    "chai": "~3.2.0",
-    "mocha": "~2.3.2",
     "bunyan": "^1.5.1",
-    "hapi": "^11.1.3"
+    "chai": "~3.2.0",
+    "hapi11": "npm:hapi@11.x.x",
```
Comment (dafortune) [inline package.json]: I didn't know this feature 😮

File: test/hapi_server.js (hunk 2 above)
Comment (dafortune) [reply test/hapi_server.js]: Might have had a 0 as default and now it has changed? Do it still works if you remove the change?

File: test/hapi_server_v17.js [hunk 9]
```diff
@@ -0,0 +1,115 @@
+const EventLogger = require('../').EventLogger;
```
Comment (dafortune) [inline test/hapi_server_v17.js]: Missing error tests? Or were they excluded on purpose?

**Comment by dafortune** [comment]
Minor things, we should also document the usage in the readme.

File: test/hapi_server.js (hunk 2 above)
Comment (joseluisdiaz) [reply test/hapi_server.js]: it's tested against each specific version of hapi!

File: eventlogger/index.js (hunk 6 above)
Comment (joseluisdiaz) [reply eventlogger/index.js]: Will be better to have one _hapi_ plugin that's uses `auth0-common-logging` that gets updated. But i think that a factory method is a reasonable way to solve it.

File: package.json (hunk 8 above)
Comment (joseluisdiaz) [reply package.json]: No, only YARN :)

File: test/hapi_server_v17.js (hunk 9 above)
Comment (joseluisdiaz) [reply test/hapi_server_v17.js]: Not sure if i follow; what do you mean?
